concordium-consensus/src/Concordium/Scheduler/Runner.hs

```diff
@@ -125,15 +127,17 @@ processUngroupedTransactions ::
   m Types.GroupedTransactions
 processUngroupedTransactions inpt = do
   txs <- processTransactions inpt
-  return (Types.fromTransactions (map ((:[]) . Types.fromAccountTransaction 0) txs))
+  -- We just attach a 'MaybeOk' to the transaction such that it will get verified by the scheduler.
+  return (Types.fromTransactions (map ((:[]) . (\x -> (Types.fromAccountTransaction 0 x, TVer.MaybeOk TVer.NormalTransactionInsufficientFunds))) txs))
```

Comment:
I think this is very bad. It's a completely made up reason.

If there is no simple way of getting rid of this, then the rationale for it really should be well-documented.

Reply:
Tranactions can now be accompanied by a `Maybe VerificationResult` instead of a `VerificationResult` when arriving at the `Scheduler`. 

This was a solution to the problem where if a transaction was included in different blocks but at the same time it was also finalized in a parent block. WIth the prior solution the block would be rejected in a wrong way and not by the proper duplicate `TxFailure`. And as such these are now constructed with `(tx, Nothing)`.